nin/backend/layer.js

```diff
@@ -0,0 +1,95 @@
+var fs = require('fs');
+
+function add(projectPath, layer, callback) {
+  read(projectPath, function (err, layers) {
+    if (err) {
+      if (callback) {
+        callback(err);
+      }
+      return;
+    }
+    var defaultLayer = {
+      type: '',
+      displayName: '',
+      startFrame: 0,
+      endFrame: 1000,
+      color: 'red',
+      config: {}
+    };
+    layers.unshift(mergeOptions(layer, defaultLayer));
+    write(projectPath, layers, function (err) {
+      if (callback) {
+        callback(err);
+      }
+    });
+  });
+}
+
+function update(projectPath, layerId, layer, callback) {
+  read(projectPath, function (err, layers) {
+    if (err) {
+      if (callback) {
+        callback(err);
+      }
+      return;
+    }
+    for (var key in layer) {
+      layers[layerId][key] = layer[key];
+    }
+    write(projectPath, layers, function (err) {
+      if (callback) {
+        callback(err);
+      }
+    });
+  });
+}
+
+var operationQueue = [];
+var active = null;
+
+function read(projectPath, callback) {
+  operationQueue.push([projectPath, callback]);
+  if (active === null) {
+    advanceQueue();
+  }
+}
+
+function write(projectPath, layers, callback) {
+  var data = JSON.stringify(layers, null, 2) + '\n';
```

Comment:
We should probably use crossplatform line-endings as this might die on iver's windows machine